integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/hive/CarbonInternalMetastore.scala

```diff
@@ -192,6 +226,42 @@ object CarbonInternalMetastore {
           LOGGER.error(e.getMessage)
       }
     }
+    if (null != indexExists) {
+      if (null != carbonTable && (null == indexExists || indexExists.toBoolean)) {
```

Comment:
indexExists is not null, that is when it enters this line.

Reply:
Handled